src/services/identity/SitesService.ts

```diff
@@ -78,6 +84,107 @@ class SitesService {
     return user.site_members.map((site) => site.repo?.name)
   }
 
+  async getCommitAuthorEmail(commit: GitHubCommitData) {
+    const { message } = commit
+
+    // Commit message created as part of phase 2 identity
+    if (message.startsWith("{") && message.endsWith("}")) {
+      const { userId }: { userId: string } = JSON.parse(message)
+      const user = await this.usersService.findById(userId)
+
+      if (user) {
+        return user.email
+      }
+    }
+
+    // Legacy style of commits, or if the user is not found
+    const {
+      author: { email: authorEmail },
+    } = commit
+    return authorEmail
+  }
+
+  async getUrlsOfSite(
+    sessionData: UserWithSiteSessionData,
+    urlTypes: SiteUrlTypes[]
+  ): Promise<SiteUrls> {
+    // Tries to get the site urls in the following order:
+    // 1. From the deployments database table
+    // 2. From the config.yml file
+    // 3. From the GitHub repository description
+    // Otherwise, returns an empty string for the url
+    const { siteName } = sessionData
```

Comment:
i think the code below is pretty janky to read and we should instead write it as a flow of something like

1. obtain data 
2. use util function to assign properties if it exists in data 
3. check if both properties are valid otherwise do 1-3 again

wdyt about this?

Reply:
Adjusted in 4432732.